client/app/telecom/telephony/line/phone/programmableKeys/telecom-telephony-line-phone-programmableKeys.controller.js

```diff
@@ -83,6 +83,84 @@ angular.module("managerApp").controller("TelecomTelephonyLinePhoneProgammableKey
         });
     };
 
+    /* ===========================
+    =            BULK            =
+    ============================ */
+
+    self.bulkDatas = {
+        infos: {
+            name: "functionKeys",
+            actions: [{
+                name: ""
+            }]
+        }
+    };
+
+    self.getBulkParams = function () {
+        self.bulkDatas.infos.actions = self.buildBulkActions();
+    };
+
+    self.buildBulkActions = function () {
+        return _.map(self.functionKeys.raw, function (key) {
+            return {
+                name: "functionKey",
+                route: `/telephony/{billingAccount}/line/{serviceName}/phone/functionKey/${key.keyNum}`,
```

Comment:
template strings are good :-) but you are mixing ES6 feature into ES5 code :-$ Can you avoid that ?